scanner/src/funTest/kotlin/HttpCacheTest.kt

```diff
@@ -76,15 +76,15 @@ class HttpCacheTest : StringSpec() {
     }
 
     override fun beforeTest(description: Description) {
-        // Start the local HTTP server.
+        // Start the local HTTP server with the system default value for queued incoming connections.
         server = HttpServer.create(InetSocketAddress(loopback, port), 0)
         server.createContext("/", MyHttpHandler())
         server.start()
     }
 
     override fun afterTest(description: Description, result: TestResult) {
-        // Ensure the server is properly stopped even in case of exceptions.
-        server.stop(0)
+        // Ensure the server is properly stopped even in case of exceptions, but wait at most 5 seconds.
+        server.stop(5)
```

Comment:
Is it actually necessary to start / stop the server before each test cases.
I'm not sure how heavy weigth this is I have to admit - if it is slow would it be sufficient to 
override beforeClass / afterClass instead?

Reply:
We should definitely do this for cleanliness. But I'm also going to further change the test to allow test to run in parallel.

Reply:
Turns out kotlintest always [runs test cases in a test spec sequentially](https://github.com/kotlintest/kotlintest/blob/master/doc/reference.md#parallelism), so no need for a change here.

Reply:
Sorry, my comment was quite off topic. Let's forget about this for now.